agent/engine/docker_task_engine_linux_test.go

```diff
@@ -1364,3 +1366,29 @@ func TestProvisionContainerResourcesBridgeModeWithServiceConnect(t *testing.T) {
 		require.Nil(t, taskEngine.(*DockerTaskEngine).provisionContainerResources(testTask, cont).Error)
 	}
 }
+
+func TestWatchAppNetImage(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.TODO())
+	defer cancel()
+	ctrl, _, _, taskEngine, _, _, _, serviceConnectManager := mocks(t, ctx, &defaultConfig)
+	defer ctrl.Finish()
+
+	// create a fsnotify watcher
+	watcher, err := fsnotify.NewWatcher()
+	assert.NoError(t, err)
+	err = watcher.Add("/tmp/")
+	assert.NoError(t, err)
+	defer watcher.Close()
+	event := fsnotify.Event{
+		Name: "/tmp/",
+		Op:   fsnotify.Create,
+	}
```

Comment:
instead of faking an event, can we try actually creating a file in `/tmp` and check that watcher is able to pick it up?

Reply:
same reason as mentioned in the previous comment. 

The unit test failed with open /tmp/serviceconnect/agent.tar : permission denied in previous revision when we tried to test it by creating a file in /tmp, my understanding is that we need to run this test with sudo permission in order for test to create files with os package(this test passed on local when run with sudo command).

I split the watchAppnetImage into two functions so that we could test the image reload and relay task restart logic by creating fake event and avoid creating any temp files.

Reply:
Try https://pkg.go.dev/testing#T.TempDir, that should help us test file operations in a directory without needing privileged access.